### ISSUE TITLE
ntLink cannot read fasta files with multi-line sequences

### ISSUE THREAD

**Issue by RaverJay** [bcgsc/ntLink#17]
My flye assembly looks like this:

```
>contig_1
CAATTTTGGTGGTCTTCAGCTCAATAGAGTCTACCAGCACAGTAGAAGAGCAGAGGGATC
CTAAACGATGAAAATCATTTAGTCAGCTGATCAGGTGACAGCACTGACGTGTCACAAAGT
CGATAGTAAACACGACAACAAACTAGCCAATAAGAGTGTTTATTTAATCATGCAATCTTT
CATTACAGGACAACTAAATTTCATCATTAAGAGTGGTAACATATGAGTATTGCAGTTAAT
>contig_2
....
```

ntLink dies with the error:
```
ntLink scaffold target=assembly.fa reads=reads.fq.gz t=5
/..../ntlink-1.0.1-0/src/indexlr --long --pos --strand -k 32 -w 100 -t 5 assembly.fa > assembly.fa.k32.w100.tsv
[2021-07-29 10:47:44][ERROR] assembly.fa source file is in invalid format!
make: *** [/home/ya86gul/miniconda3/envs/cricket/bin/share/ntlink-1.0.1-0/ntLink:141: assembly.fa.k32.w100.tsv] Error 1```

The used sequence reader seems to not support multi-line sequence fasta files

Best

**Comment by lcoombe** [comment]
Hi @RaverJay,

Yes, you are correct, the sequence reader used by ntLink expects the input fasta files to be single-line, not multi-line. When I need to do this conversion (from multi-line fasta to single-line), I just do a quick `seqtk seq` command (or many other one-liners will work). 
I will also add a note about this to the README if others have the same question.

Thanks for your interest in ntLink!
Lauren

**Comment by RaverJay** [comment]
Yeah the conversion is easy enough, just thought you might wanna add the capability =)

**Comment by lcoombe** [comment]
For sure - I'll put that on our list of future TODOs!

**Comment by lcoombe** [comment]
@schutzekatze - is this a feature that you could add to the btllib SeqReader?

**Comment by schutzekatze** [comment]
@lcoombe Yeah, it's a planned feature. I'll see to add it these days.

**Comment by lcoombe** [comment]
Great - thanks Vlad!
And thanks @RaverJay for giving us the nudge to address this!

**Comment by lcoombe** [comment]
Hey @schutzekatze - any update on this feature?

**Comment by schutzekatze** [comment]
It's implemented in the latest btllib master branch, but I was waiting to finish another feature before making a release. We could add the latest btllib to ntLink to get this feature right now.

**Comment by lcoombe** [comment]
Sounds good - no rush, we can definitely wait until the next btllib release.

**Comment by vlad0x00** [comment]
Support for this has been added to the master branch and will be available in the next release.
